### DISCUSSION TITLE
Visualising/reporting on how quickly packages adopt new versions of Swift

### DISCUSSION THREAD

**Discussion by Sherlouk** [SwiftPackageIndex/SwiftPackageIndex-Server#1350] [Ideas]
In #438 I proposed the following:

> I want to know if a package is slow to adopt new swift versions

I think this is a very interesting challenge but wasn't really in scope for the changes being discussed there so breaking it out though there is some overlap. It's an important decision I (at least) take into account when choosing libraries in order to not block myself in the future.

One idea I had was if we store the swift version of a library for each version (possibly covered by #303/#191) we would be able to compare when a new swift version was released and when a package's swift version updated to be that version.

Example:
```
1st Jan: Package 1's Swift Version = 1.0
3rd Jan: Swift releases 2.0
10th Jan: Package 1's Swift Version = 2.0
```
We know that Package 1 went from 1.0 to 2.0, and the delta is 7 days since Swift 2.0 was released.

Not sure how we could display this information, in an information priority list this is towards the bottom but I do think it's valuable information. Would be interested with what other people think?

Relates to #371

**Comment by Sherlouk** [comment]
Maybe part of this is also to highlight (where we currently display what swift version is supported) whether that's the latest version? I wonder if there's a data source we can use to get that information.

**Comment by daveverwer** [comment]
Yea, we have big plans in this area, and the data gathering mechanism to make this possible is actively being worked on right now as our next major feature.

I'm also working on the design for it, but haven't got anything worth showing yet. This is certainly a very important metric though.
